src/App.tsx

```diff
@@ -13,49 +13,54 @@ import {
   Text,
   VStack,
 } from "@chakra-ui/react";
-
+import usePublisher from "./hooks/usePublisher";
 import IconCamera from "./components/Icons/Camera";
 import IconCameraOff from "./components/Icons/CameraOff";
+import type { Publisher, PublisherState } from './hooks/types/Publisher';
 
-type PublishState = "Setup" | "Connecting" | "Ready" | "Streaming";
 
 interface AppState {
-  publishState: PublishState;
+  publishState: PublisherState;
   mediaStream?: MediaStream;
 }
 
 type PublishAction = "join" | "joined" | "goLive" | "stopLive";
 
 function App() {
   const initialState: AppState = {
-    publishState: "Setup",
+    publishState: "ready",
   };
 
   const [shouldRecord, setShouldRecord] = useState(false);
   const [cameraOn, setCameraOn] = useState(true);
   const [participantsCount] = useState(0);
 
+  const token = import.meta.env.VITE_MILLICAST_STREAM_PUBLISHING_TOKEN;
+  const streamId = import.meta.env.VITE_MILLICAST_STREAM_NAME;
+
+  const { startStreaming, stopStreaming, publisherState } = usePublisher(token, streamId);
+
   const reducer = (state: AppState, action: { type: PublishAction }) => {
     switch (action.type) {
       case "join": {
         // TODO: connect publisher
-        const newState: AppState = { ...state, publishState: "Connecting" };
+        const newState: AppState = { ...state, publishState: "connecting" };
```

Comment:
We don't need to use AppState and Actions here anymore, only need the state returned by using the hook

Reply:
I think you're reviewing an old file. There is no more reducer in App.tsx